src-tauri/src/farmer.rs

```diff
@@ -60,7 +61,22 @@ async fn farm(
     plot_size: u64,
     error_sender: Sender<()>,
 ) -> Result<(), anyhow::Error> {
+    if let Err(error) = timeout(Duration::from_secs(20), async {
+        loop {
+            if TcpStream::connect("127.0.0.1:9947").is_ok() {
```

Comment:
Just a minor thing, but it would be better to use `tokio::net::TcpStream` here, as `std` one blocks future

Reply:
thanks!